@kiva/kv-components/vue/KvLightbox.vue

```diff
@@ -0,0 +1,285 @@
+								tw-pb-2.5 md:tw-pb-4
+								tw-overflow-auto
+							"
+						>
+							<!-- @slot default -->
+							<slot>Lightbox body</slot>
+						</div>
+
+						<!-- controls -->
+						<div
+							v-if="$slots.controls"
+							ref="controlsRef"
+							class="
+								tw-flex-shrink-0 tw-flex tw-justify-end tw-gap-x-2.5
+								tw-p-2.5 md:tw-px-4 md:tw-pb-4 md:tw-pt-1
+							"
+						>
+							<!-- @slot controls -->
+							<slot name="controls"></slot>
+						</div>
+					</div>
+				</div>
+			</focus-lock>
+		</div>
+	</transition>
+</template>
+
+<script>
+
+import { mdiClose } from '@mdi/js';
+import FocusLock from 'vue-focus-lock';
+import { hideOthers as makePageInert } from 'aria-hidden';
+import { lockScroll, unlockScroll } from '../utils/scrollLock';
+import { lockPrintSingleEl, unlockPrintSingleEl } from '../utils/printing';
+
+import KvMaterialIcon from './KvMaterialIcon.vue';
+
+/**
+ * Alert or a lightbox
+ * Accesibility: https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal
+ *
+ * - [x] Tab and Shift + Tab do not move focus outside the dialog
+ * - [x] focus is initially set on the first focusable element (close button).
+ * - [x] focus is returned to the element that opened the dialog on close
+ * - [x] role = dialog
+ * - [x] aria-label is set to its title.
+ * - [x] adds aria-hidden=true to all elements other than this dialog when open. - https://github.com/theKashey/vue-focus-lock/issues/16
+ * - [x] scrolling only scrolls the lightbox contents, not the page itself
+
+ * Alert dialog - https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_alertdialog_role
+ *
+ * - [x] focus moves to the first non-destructive control, rather than the close button
+ * - [x] role = alertDialog
+ * - [x] aria-describedby is set to the id of the dialog body
+ *
+ * Printing
+ *
+ * - [x] Only prints the contents of the lightbox when open
+ * TODO: ensure purge doesn't remove these classes
+ * 'print:tw-invisible',
+ * 'print:tw-visible',
+ * 'print:tw-overflow-auto',
+ * 'print:tw-overflow-hidden',
+ */
+
+export default {
+	components: {
+		FocusLock,
+		KvMaterialIcon,
+	},
+	props: {
+		/**
+		 * Whether the dialog is open or not
+		 * */
+		visible: {
+			type: Boolean,
+			default: false,
+		},
+		/**
+		 * Appearance and role of the lightbox
+		 * @values lightbox, alert
+		 * */
+		variant: {
+			type: String,
+			default: 'lightbox',
+		},
+		/**
+		 * The title of the dialog which describes the dialog to screenreaders, and if no
+		 * content is in the `header` slot, will be displayed at the top of the lightbox.
+		 * */
+		title: {
+			type: String,
+			required: true,
+		},
+		/**
+		 * The dialog has no close X button, clicking the screen does not close,
+		 * pressing ESC does not close.
+		 * */
+		preventClose: {
+			type: Boolean,
+			default: false,
+		},
+	},
+	data() {
+		return {
+			mdiClose,
+			hideOthers() {}, // reference to aria-hide function
```

Comment:
I don't see `hideOthers` used anywhere. Is this supposed to be `makePageInert`?

Reply:
Yes, missed it in a refactor. Thanks!